pumpkin-world/src/level.rs

```diff
@@ -189,16 +176,22 @@ impl Level {
         self.chunk_watchers.shrink_to_fit();
     }
 
-    pub fn write_chunk(&self, _chunk_to_write: (Vector2<i32>, Arc<RwLock<ChunkData>>)) {
-        //TODO
+    pub async fn write_chunk(&self, chunk_to_write: (Vector2<i32>, Arc<RwLock<ChunkData>>)) {
```

Comment:
We need some kind of file lock so multiple threads don't try to write to the same file at the same time/read while the file is being written